### PR TITLE
davinci-resolve: 18.5.1 -> 18.6.2; add studio variant; various improvements

### PR DESCRIPTION
**Pull request by amarshall**
## Description of changes

See individual commit messages for details and caveats.

## Things done

Tested both free and studio. Tested studio activation. Tested that app starts and can load existing project and various functionality. Using Nvidia GPU.

The Studio download is a bit flaky and sometimes fails with

> curl: (92) HTTP/2 stream 1 was not closed cleanly: INTERNAL_ERROR (err 2)

but works eventually.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/davinci-resolve/default.nix [hunk 1]
```diff
@@ -47,19 +50,24 @@ let
...
           # ENV VARS
           SSL_CERT_FILE = "${cacert}/etc/ssl/certs/ca-bundle.crt";
 
           # Get linux.downloadId from HTTP response on https://www.blackmagicdesign.com/products/davinciresolve
-          DOWNLOADID = "defc1c6789b7475b9ee4a42daf9ba61d";
           REFERID = "263d62f31cbb49e0868005059abcb0c9";
-          SITEURL = "https://www.blackmagicdesign.com/api/register/us/download/${DOWNLOADID}";
+          DOWNLOADSURL = "https://www.blackmagicdesign.com/api/support/us/downloads.json";
+          SITEURL = "https://www.blackmagicdesign.com/api/register/us/download";
+          PRODUCT = "DaVinci Resolve${if studioVariant then " Studio" else ""}";
```
Comment (stelcodes) [inline pkgs/applications/video/davinci-resolve/default.nix:69]: Same here, `lib.optionalString`

File: pkgs/applications/video/davinci-resolve/default.nix [hunk 2]
```diff
@@ -22,13 +22,16 @@
...
+
+, studioVariant ? false
 }:
 
 let
   davinci = (
     stdenv.mkDerivation rec {
-      pname = "davinci-resolve";
-      version = "18.5.1";
+      pname = "davinci-resolve${if studioVariant then "-studio" else ""}";
```
Comment (stelcodes) [inline pkgs/applications/video/davinci-resolve/default.nix:33]: Better to use `lib.optionalString` here

**Review by stelcodes — COMMENTED** [review]
A couple comments

**Review by jshcmpbll — APPROVED** [review]

**Comment by xvrqt** [comment]
I bought a license to try studio version;  it builds fine, but upon running it complains that it can't copy the license file into the read-only `/nix/store` path (makes sense). 

It seems like a recreation of an older issue: https://github.com/NixOS/nixpkgs/issues/94032#issuecomment-921918602 where davinci-resolve reverts to its `/nix/store` location no matter where it is run from. This was seeming solved by wrapping it in a FHS location built for davinci using https://github.com/NixOS/nixpkgs/blob/bcd452a5544131d486eac0493002d2341b3af022/pkgs/build-support/build-fhs-userenv-bubblewrap/default.nix. 

The issue was brought up again in https://github.com/NixOS/nixpkgs/pull/152113#issuecomment-1075642604 where the work around was to use `extraBwrapArgs` to bind the license file to a writeable location in the FHS. However this is failing for me :[ 

I am installing it via a standalone home-manager installation, and pulling nixpkgs-unstable via a flake.
Installing it as a `systemPackages` on my NixOS configuration produces the same error.

`bwrap: Can't create file at /nix/store/kxy0cl1p0f9nr6gf46gzc1jx42f48kzw-davinci-resolve-studio-18.6.3/.license: Is a directory`

I'm pretty green to NixOS so I'm trying to understand `buildFHSEnv` better to see why davinci isn't running from there, but would appreciated if someone more knowledgeable would take a look.

**Comment by xvrqt** [comment]
Seems related: https://github.com/NixOS/nixpkgs/issues/239017

Yet, we're clearly using `--bind` in the `extraBwrapArgs` so idk what's up :[

**Pull request merged**